code.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    while i < self.length { |i| yield(self[i]) }
```

Comment:
Did you test this method? :thinking: 

You seem not to be using `while` the right way. 
Also, note that in some cases you don't need to call `self`

As suggested earlier, please test the methods with [these examples](https://ruby-doc.org/core-2.6.4/Enumerable.html)

This method should also return `Enumerator` when no block is given.
:bulb: investigate the `to_enum` and `block_given?` methods.